pandas/core/missing.py

```diff
@@ -490,6 +491,10 @@ def interpolate_2d(
     if ndim == 1:
         values = values[0]
 
+    if orig_values.dtype.kind == "M":
+        # convert float back to datetime64
+        values = values.astype(orig_values.dtype)
```

Comment:
Can this fail if not all the values are filled (e.g. with `limit`)?

Reply:
I don't see why it would.  This is functionally the same as what we're doing now, just in a more-correct place.

Reply:
this likely should be for .kind in ['M', 'm']

Reply:
or even better this should be is_extension_dtype then cast back

Reply:
its not tz case we're catching here, it's regular datetime64

Reply:
k